ports/psoc6/machine_i2s.c

```diff
@@ -366,21 +323,6 @@ static void mp_machine_i2s_init_helper(machine_i2s_obj_t *self, mp_arg_val_t *ar
         mp_raise_ValueError(MP_ERROR_TEXT("invalid format"));
     }
 
-    // is valid clock freq?
-    uint32_t audio_clock_freq_hz;
-    uint32_t rate = args[ARG_rate].u_int;
```

Comment:
We should keep this part checking that the rate is a valid one or?
Otherwise, the use could pass any integer. But actually only the ones filtered where valid?

Additionally, here I would check if the clock is properly for that rate, and raise exception if that is not the case.